eth/stagedsync/stage_headers.go

```diff
@@ -286,3 +279,76 @@ Error: %v
 	log.Info(fmt.Sprintf("[%s] Imported new block headers", logPrefix), ctx...)
 	return newCanonical, reorg, forkBlockNumber, nil
 }
+
+func verifyHeaders(db ethdb.Database, engine consensus.EngineAPI, headers []*types.Header, seals []bool) error {
+	toVerify := len(headers)
+	if toVerify == 0 {
+		return nil
+	}
+
+	requests := make(chan consensus.VerifyHeaderRequest, 1)
+	id := rand.Uint64()
+	requests <- consensus.VerifyHeaderRequest{id, headers, seals, nil}
```

Comment:
Instead of using `results` channel, could you simply do `engine.HeaderVerification() <- consensus.VerifyHeaderRequest{id, headers, seals, nil}` here and remove the corresponding clause from `select` below?